cpp/src/arrow/util/benchmark_util.h

```diff
@@ -15,10 +15,24 @@
 // specific language governing permissions and limitations
 // under the License.
 
+#include <condition_variable>
+#include <mutex>
```

Comment:
Ah, this is going to be a problem.  We can't include `mutex` in public headers due to a Windows compatibility issue.  Can you create a `benchmark_util.cc` and move the definition of these methods there?

Reply:
@westonpace , sorry if this is trivial, but I'm having some trouble here.

I committed some changes, but it seems like the compiler is having trouble finding third party dependencies imported in `benchmark_util.h` (like benchmarks, gtest) after including `benchmark_util.cc` in the `src/arrow/CMakeLists.txt`. Let me know if I'm missing a step.

Reply:
Hi @westonpace, have you had a chance to take a look at this?